compiler-k2/src/test/kotlin/com/squareup/anvil/compiler/k2/fir/contributions/ContributesBindingFirExtensionTest.kt

```diff
@@ -0,0 +1,101 @@
+package com.squareup.anvil.compiler.k2.fir.contributions
+
+import com.google.common.truth.Truth.assertThat
+import com.squareup.anvil.compiler.internal.fqName
+import com.squareup.anvil.compiler.testing.CompilationMode
+import com.squareup.anvil.compiler.testing.CompilationModeTest
+import com.squareup.anvil.compiler.testing.compilation.Compile2Result
+import com.squareup.anvil.compiler.testing.contributingObject
+import com.squareup.anvil.compiler.testing.parentInterface
+import dagger.Binds
+import org.junit.jupiter.api.TestFactory
+
+class ContributesBindingFirExtensionTest : CompilationModeTest(
+  CompilationMode.K2(useKapt = false),
+) {
+
+  @TestFactory
+  fun `generated class contributes to expected scope`() = testFactory {
+    compile2(
+      """
+      package com.squareup.test.other
+
+      interface AppScope
+      """.trimIndent(),
+      """
+      package com.squareup.test
+
+      import com.squareup.anvil.annotations.ContributesBinding
+      import com.squareup.test.other.AppScope
+      import javax.inject.Inject
+      
+      interface ParentInterface
+
+      @ContributesBinding(
+        scope = AppScope::class, 
+        boundType = ParentInterface::class
+      )
+      object ContributingObject : ParentInterface
+      """.trimIndent(),
+    ) {
+      val contributesToScope = contributingObjectBindingModule.contributesToAnnotation.scope
+
+      assertThat(contributesToScope.fqName.asString()).isEqualTo("com.squareup.test.other.AppScope")
+    }
+  }
+
+  @TestFactory
+  fun `generated class is a dagger module`() = testFactory {
+    compile2(
+      """
+      package com.squareup.test
+
+      import com.squareup.anvil.annotations.ContributesBinding
+      import javax.inject.Inject
+      
+      interface ParentInterface
+
+      @ContributesBinding(
+        scope = Int::class, 
+        boundType = ParentInterface::class
+      )
+      object ContributingObject : ParentInterface
+      """.trimIndent(),
+    ) {
+      // This is an implicit assertion and will fail if the annotation is missing
+      contributingObjectBindingModule.daggerModule
```

Comment:
```suggestion
      classGraph.shouldContainClass(CommonNames.contributingObject.bindingModule)
```

That `shouldContainClass` should be infix.  That's my bad.  The `bindingModule` extension doesn't exist, but it should.

Reply:
Similar thoughts here as my response to the kotest comment for the syntactic changes, but would you like to still see a more explicit assert instead of the implict one?

Reply:
It's not the end of the world, but yes.  It's one line that's clear about its purpose and gives an `AssertionError` as its exception.